src/sql/store.py

```diff
@@ -98,14 +99,23 @@ def __init__(self, store: SQLStore, query: str, with_: Iterable = None):
             )
 
     def __str__(self) -> str:
-        if self._query:
-            with_all = _get_dependencies(self._store, self._with_)
-            parsed_res = parse_one(self._query)
-            for with_key in with_all:
-                with_query = self._store._data[with_key]._query
-                parsed_res = parsed_res.with_(with_key, with_query)
-            return str(parsed_res.sql())
-        return ""
+        with_template = Template(
+            """WITH{% for name in with_ %} "{{name}}" AS ({{saved[name]._query}})\
+{{ "," if not loop.last }}{% endfor %}{{query}}"""
+        )
+        with_template_backtick = Template(
+            """WITH{% for name in with_ %} `{{name}}` AS ({{saved[name]._query}})\
+{{ "," if not loop.last }}{% endfor %}{{query}}"""
+        )
+        is_use_backtick_template = (
+            sql.connection.Connection._is_curr_dialect_support_backtick()
+        )
+        with_all = _get_dependencies(self._store, self._with_)
+        print("with_all", with_all)
+        template = with_template_backtick if is_use_backtick_template else with_template
```

Comment:
As I understand, this PR should handle 2 templates and dynamically choose the right one. I guess there are more possible templates we might handle in the future, so to make it easier to add more templates in the future I think we can export the template logic to another function i.e : `_get_template()` which returns the right template for a given connection.
```
def __str__(self) -> str:
        template = _get_template()
        return template.render(
            query=self._query, saved=self._store._data, with_=with_all
        )
```

as for the implementation of choosing the right template, I think at this point, when we have only 2 possible templates - it's easy to use the if statement and manage 2 queries, so we can leave it as is.

But we need to keep in mind that if we'll want to add another template we'll have to implement another function like `is_use_backtick_template` i.e : `is_use_something_new_template`.

So, I think if there is not much effort in making it more scalable to handle future templates - I would do it (maybe a dynamic query builder with given identifiers).
If not - I think we can add a comment explaining what are the steps to create a new template.

Reply:
Yes, I agree your point!
We can have the helper class to handle this in the future if there are more templates needed.